compute_tools/src/bin/compute_ctl.rs

```diff
@@ -349,11 +423,47 @@ fn main() -> Result<()> {
         }
     }
 
+    Ok((
+        pg,
+        StartPostgresResult {
+            delay_exit,
+            compute,
+            #[cfg(target_os = "linux")]
+            rt,
+            #[cfg(target_os = "linux")]
+            token,
+            #[cfg(target_os = "linux")]
+            vm_monitor,
+        },
+    ))
+}
+
+type PostgresHandle = (std::process::Child, std::thread::JoinHandle<()>);
+
+struct StartPostgresResult {
+    delay_exit: bool,
+    // passed through from WaitSpecResult
+    compute: Arc<ComputeNode>,
+
+    #[cfg(target_os = "linux")]
+    rt: Option<tokio::runtime::Runtime>,
+    #[cfg(target_os = "linux")]
+    token: tokio_util::sync::CancellationToken,
+    #[cfg(target_os = "linux")]
+    vm_monitor: Option<tokio::task::JoinHandle<Result<()>>>,
+}
+
+fn wait_postgres(
+    pg: Option<PostgresHandle>,
+    startup_context_guard: Option<opentelemetry::ContextGuard>,
+) -> Result<WaitPostgresResult> {
     // Wait for the child Postgres process forever. In this state Ctrl+C will
     // propagate to Postgres and it will be shut down as well.
     let mut exit_code = None;
     if let Some((mut pg, logs_handle)) = pg {
         // Startup is finished, exit the startup tracing span
+        // TODO: Probably easier to drop startup_context_guard outside this function. It's here
+        // right now because keeping it here reduced the size of the diff.
```

Comment:
Doing this in a future commit within this PR?

Reply:
Not sure what semantics are intended around the startup context — initially decided to leave it for now pending any discussion that should be had on that

But otherwise happy to do so.

Reply:
This function doesn't do anything with it, except drop it, so +1 for moving it to the caller.

Reply:
See https://github.com/neondatabase/neon/pull/7600 for suggestion on how that would look

Reply:
Implemented #7600 via 8284db01f6ff12415421f5f820611143b66bc9f9 and ffe65ef84f386d3bc4b34cf41e2c3212aee624c9. Some adjustments made to avoid moving things around (because it makes the diff big, which makes merge conflicts harder to resolve)